#!/usr/bin/env bash

#PBS -l select=1:ncpus=2:mem=4gb
#PBS -l walltime=12:00:00
#PBS -j oe

# Move to directory where qsub was called
cd $PBS_O_WORKDIR

# Load nextflow module and katana-specific fixes
module add java/11.0.17_8-openjdk nextflow/24.04.4
export PATH="$(realpath ./bin):$PATH"
IGENOMES='/data/bio/igenomes/references'

# Pull latest
nextflow pull nf-core/rnaseq

# Start pipeline
nextflow run nf-core/rnaseq     `# Name of nf-core pipeline` \
    -params-file nf-params.json `# Params file (edited by you)` \
    -resume                     `# Outdated, but allows specifying e.g. hg38 in nf-params.json` \
    -r 3.17.0                   `# Could be outdated, but good practice` \
    --igenomes_base $IGENOMES   `# Outdated/for testing, allows specifying e.g. "genome": "GRCh37" in nf-params.json instead of fasta/gtf` \
    --publish_dir_mode 'link'   `# Try to link output files from work dirs to save space` \